index.js

```diff
@@ -1,22 +1,56 @@
-module.exports = Hook
-
 var register = require('./lib/register')
 var addHook = require('./lib/add')
 var removeHook = require('./lib/remove')
 
-function Hook () {
+// https://stackoverflow.com/a/21792913
```

Comment:
maybe above the stackoverflow link, add a summary so I only need to open the URL when I look for more details?

```suggestion
// bind with array of arguments
// https://stackoverflow.com/a/21792913
```

Reply:
I've added that line and combined it with the link, less lines 👍